platforms/ide/ide/src/main/java/org/gradle/plugins/ide/internal/tooling/ResilientGradleBuildBuilder.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2013 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.plugins.ide.internal.tooling;
+
+import org.gradle.api.GradleException;
+import org.gradle.api.initialization.ProjectDescriptor;
+import org.gradle.api.internal.GradleInternal;
+import org.gradle.api.internal.SettingsInternal;
+import org.gradle.internal.build.BuildState;
+import org.gradle.internal.build.BuildStateRegistry;
+import org.gradle.internal.composite.BuildIncludeListener;
+import org.gradle.internal.problems.failure.Failure;
+import org.gradle.plugins.ide.internal.tooling.model.BasicGradleProject;
+import org.gradle.plugins.ide.internal.tooling.model.DefaultGradleBuild;
+import org.gradle.plugins.ide.internal.tooling.model.DefaultResilientGradleBuild;
+import org.gradle.tooling.internal.gradle.DefaultProjectIdentifier;
+import org.jspecify.annotations.NullMarked;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@NullMarked
+public class ResilientGradleBuildBuilder extends GradleBuildBuilder {
+    private final BuildIncludeListener failedIncludedBuildsRegistry;
+    private Map<BuildState, Failure> brokenBuilds = new HashMap<>();
+    private Map<SettingsInternal, Failure> brokenSettings = new HashMap<>();
+
+    public ResilientGradleBuildBuilder(
+        BuildStateRegistry buildStateRegistry,
+        BuildIncludeListener failedIncludedBuildsRegistry
+    ) {
+        super(buildStateRegistry);
+        this.failedIncludedBuildsRegistry = failedIncludedBuildsRegistry;
+    }
+
+    @Override
+    public boolean canBuild(String modelName) {
+        return modelName.equals("org.gradle.tooling.model.gradle.ResilientGradleBuild");
+    }
+
+    @Override
+    protected void ensureProjectsLoaded(BuildState target) {
+        try {
+            target.ensureProjectsLoaded();
+        } catch (GradleException e) {
+            if (e.getCause() instanceof org.gradle.kotlin.dsl.support.ScriptCompilationException) {
```

Comment:
❓ When commenting the last time I was thinking we would try to handle any Gradle exception (e.g. to also runtime errors) and not just `ScriptCompilationException`. I see we still then conditionally handle just `ScriptCompilationException` in the end. Is there a different behaviour if we try to do handle also runtime exception?

We can also merge this atm, and try to handle runtime errors in a PR later